cvat/apps/dataset_manager/bindings.py

```diff
@@ -1149,9 +1153,23 @@ def convert_attrs(label, cvat_attrs):
                 z_order=shape_obj.z_order)
         elif shape_obj.type == ShapeType.RECTANGLE:
             x0, y0, x1, y1 = anno_points
-            anno = datum_annotation.Bbox(x0, y0, x1 - x0, y1 - y0,
-                label=anno_label, attributes=anno_attr, group=anno_group,
-                z_order=shape_obj.z_order)
+            if shape_obj.rotation and format_name == 'Segmentation mask':
+                [cx, cy] = [(x0 + (x1 - x0) / 2), (y0 + (y1 - y0) / 2)]
+                angle = math.radians(shape_obj.rotation)
+                def rotate_point(p):
+                    [x, y] = p
+                    rx = cx + math.cos(angle) * (x - cx) - math.sin(angle) * (y - cy)
+                    ry = cy + math.sin(angle) * (x - cx) + math.cos(angle) * (y - cy)
+                    return rx, ry
+
+                poly_points = list(chain.from_iterable(map(rotate_point, [(x0, y0), (x1, y0), (x1, y1), (x0, y1)])))
+                anno = datum_annotation.Polygon(poly_points,
+                    label=anno_label, attributes=anno_attr, group=anno_group,
+                    z_order=shape_obj.z_order)
```

Comment:
We have several mask formats, which all should use this logic, but this function doesn't seem like a proper place for format checking. Consider creating an `ItemTransform`, which converts polygons with `rotation` attribute to rotated polygons - see example in https://github.com/openvinotoolkit/cvat/blob/develop/cvat/apps/dataset_manager/formats/icdar.py#L68-L85

Reply:
> but this function doesn't seem like a proper place for format checking

Before you said:
https://github.com/openvinotoolkit/cvat/blob/develop/cvat/apps/dataset_manager/bindings.py#L1088-L1169

Reply:
@zhiltsov-max Reworked as an example.

But in this case we always need to rewrite attribute "rotation" (which may exist as it was said before). 
So, maybe adding ``rotation`` property to  datum_annotation._Shape is a solution?

And what other formats use mask? (You said not only ``Segmentation mask``)

Reply:
> And as I can see, there is other format check in this function: `if format_name == "sly_pointcloud"`:

Agree, it is one of the hacks introduced with 3d support. I hope, it will be refactored at some day. We've already made some progress in this and removed few other hacks.

> So, maybe adding rotation property to `datum_annotation._Shape` is a solution?

This would seem like it is a core functionality, which is supported everywhere (in all formats etc.), while it's not. Thus, we prefer keep CVAT properties as attributes.

> But in this case we always need to rewrite attribute `rotation` (which may exist as it was said before).

It is an open question to be resolved. Maybe we can check if such attribute is defined in the label configuration and rotate shapes during mask conversion only if there is no declared attribute? We're just inventing a "reasonable" solution here, because we speak about formats that don't support such option. Maybe, we should ask about this from the customer, who wanted this feature?

> And what other formats use mask? (You said not only `Segmentation mask`)

They are listed [here](https://openvinotoolkit.github.io/cvat/docs/manual/advanced/annotation-with-polygons/creating-mask/#creating-masks). You can also [search the codebase by `dataset.transform('polygons_to_masks')`](https://github.com/openvinotoolkit/cvat/search?q=dataset.transform%28%27polygons_to_masks%27%29) to be sure. I found 7 formats.

Reply:
Added the transformation to these 7 formats. Works fine except of MOTS, OpenImages (empty masks at least if look at them using only eyes, but the transformation is called), and ICDAR Segmentation: 

![image](https://user-images.githubusercontent.com/40690378/142040468-b73b2bb1-4781-4396-b6a1-8bb9870199f8.png)

I suspect there are some format specific reasons of such behaviour, because the same results on develop branch.

Reply:
> MOTS, OpenImages (empty masks at least if look at them using only eyes, but the transformation is called)

It's ok, they have grayscale masks.

> I suspect there are some format specific reasons of such behaviour, because the same results on develop branch.

Yes, this format expects a string `color` attribute for each annotation.

@kirill-sizov , probably, this format deserves a better error message and, maybe, better handling of color info (autogeneration?).

Reply:
@zhiltsov-max, resolved it in Datumaro [PR 556](https://github.com/openvinotoolkit/datumaro/pull/556)